src/components/job/JobList.jsx

```diff
@@ -59,10 +110,213 @@ export default function JobList() {
           );
         })
       : "Loading";
-  }, [data, jobId]);
+  }, [filteredJobs, jobId]);
 
   return (
-    <div className="w-full">
+    <div className="relative flex flex-col h-screen  bg-yellow-100">
+      {/* Search bar and filter icon */}
+      <div className="flex flex-row items-center justify-between h-16 p-3 bg-blue-100">
+        <div className="relative text-gray-600">
+          <input
+            className="w-64 border-2 border-gray-300 bg-white h-10 px-5 pr-10 rounded-lg text-sm focus:outline-none"
+            type="search"
+            name="search"
+            placeholder="Search"
+            ref={searchRef}
+            onKeyDown={(e) => {
+              if (e.key === "Enter") setSearch(searchRef.current.value);
+            }}
+          />
+          <button
+            className="absolute right-0 top-1.5  mr-4 "
+            onClick={() => {
+              setSearch(searchRef.current.value);
+            }}
+          >
+            <SearchIcon />
+          </button>
+        </div>
+        <button
+          className={
+            _.isEqual(filter, emptyFilter)
+              ? "btn btn-circle btn-ghost"
+              : "btn btn-circle btn-primary"
+          }
+          onClick={() => setShowFilter(!showFilter)}
+        >
+          <FilterAltIcon fontSize="large" />
+        </button>
+      </div>
+
+      {/* Filter Panel UI */}
+      {showFilter ? (
```

Comment:
If this filter panel will only change when `showFIlter` changes, make the panel a `useMemo` that depends on `showFilter`
otherwise, this panel will re-render every time when you do the search, see a job and save a job

Reply:
I tried but with useMemo the RESET button won't work properly, so I just kept it like that